packages/reactivity-transform/src/reactivityTransform.ts

```diff
@@ -594,6 +594,79 @@ export function transformAST(
     }
   }
 
+  /**
+   * unwrap the code form the macro($、$$), fix #6312 and keep the desired behavior from RFC#369
+   */
+  function unwrapMacro(node: CallExpression) {
+    const bracketStart = node.callee.end! + offset
+    s.remove(node.callee.start! + offset, bracketStart)
+
+    if (node.arguments.length === 1) {
+      const bracketEnd = node.arguments.at(-1)!.end! + offset
```

Comment:
A side note: `Array.prototype.at` is only available in Node 16.6+.

Even in Node-only code branches we need to be conservative about using new language features since this can implicitly break user builds running on Node versions below 16.6.